modules/rtdModule/index.js

```diff
@@ -159,130 +180,142 @@ export function init(config) {
-  let dataReceived = {};
-  let processDone = false;
-  const dataWaitTimeout = setTimeout(done, _moduleConfig.auctionDelay || _moduleConfig.timeout || DEF_TIMEOUT);
-  subModules.forEach(sm => {
-    sm.getData(adUnits, onDataReceived.bind(sm));
+export function setBidRequestsData(fn, reqBidsConfigObj) {
+  _userConsent = getConsentData();
+
+  const relevantSubModules = subModules.filter(sm => typeof sm.getBidRequestData === 'function');
+  const prioritySubModules = relevantSubModules.filter(sm => !!(sm.config && sm.config.waitForIt));
+  const shouldDelayAuction = prioritySubModules.length && _moduleConfig.auctionDelay && _moduleConfig.auctionDelay > 0;
+  let callbacksExpected = prioritySubModules.length;
+  let isDone = false;
+  let waitTimeout;
+
+  if (!relevantSubModules.length) {
+    return exitHook();
+  }
+
+  if (shouldDelayAuction) {
+    waitTimeout = setTimeout(exitHook, _moduleConfig.auctionDelay);
+  }
+
+  relevantSubModules.forEach(sm => {
+    sm.getBidRequestData(reqBidsConfigObj, onGetBidRequestDataCallback.bind(sm), sm.config, _userConsent)
   });
 
-  function onDataReceived(data) {
-    if (processDone) {
-      return
+  if (!shouldDelayAuction) {
+    return exitHook();
+  }
+
+  function onGetBidRequestDataCallback() {
+    if (isDone) {
+      return;
     }
-    dataReceived[this.name] = data;
-    if (shouldWaitForAllSubModules || (this.config && this.config.waitForIt)) {
-      callbacksExpected--
+    if (this.config && this.config.waitForIt) {
+      callbacksExpected--;
     }
     if (callbacksExpected <= 0) {
-      clearTimeout(dataWaitTimeout);
-      done();
+      return exitHook();
     }
   }
 
-  function done() {
-    processDone = true;
-    callback(dataReceived);
+  function exitHook() {
+    isDone = true;
+    clearTimeout(waitTimeout);
+    fn.call(this, reqBidsConfigObj);
   }
 }
 
 /**
- * run hook after bids request and before callback
- * get data from provider and set key values to primary ad server
- * @param {function} next - next hook function
- * @param {AdUnit[]} adUnits received from auction
+ * loop through configured data providers If the data provider has registered getTargetingData,
+ * call it, providing ad unit codes, consent data and module params
+ * the sub mlodle will return data to set on the ad unit
+ * this function used to place key values on primary ad server per ad unit
+ * @param {Object} auction object received on auction end event
  */
-export function setTargetsAfterRequestBids(next, adUnits) {
-  getProviderData(adUnits, (data) => {
-    if (data && Object.keys(data).length) {
-      const _mergedData = deepMerge(setDataOrderByProvider(subModules, data));
-      if (Object.keys(_mergedData).length) {
-        setDataForPrimaryAdServer(_mergedData);
-      }
-    }
-    next(adUnits);
-  });
-}
+export function getAdUnitTargeting(auction) {
+  const relevantSubModules = subModules.filter(sm => typeof sm.getTargetingData === 'function');
```

Comment:
@omerBrowsi - this line is failing unit test. subModules at this point in the unit test is 

```
[{"name":"validSMWait","config":{"name":"validSMWait","waitForIt":true}},{"name":"validSM","config":{"name":"validSM","waitForIt":false}}]
```